upgrade.ts

```diff
@@ -0,0 +1,127 @@
+import { resolveDirectiveNameFromSelector } from './src/facade/lang';
+import { PrecompiledDirectiveFactory } from './src/core/util/bundler';
+import { createBootstrapFn } from './src/platform/browser_utils';
+import { ProviderLiteral } from './src/core/di/provider_util';
+import { reflector } from './src/core/reflection/reflection';
+
+/**
+ * `UgradeAdapterRef` controls a hybrid AngularJS v1 / Angular v2 application,
+ * but we don't have a use for it right now so no point in creating an interface for it...
+ */
+export type UgradeAdapterRef = void;
+
+export interface UpgradeAdapter {
+  new (): upgradeAdapterInstance;
+}
+
+export interface upgradeAdapterInstance {
+  /**
+   * Allows Angular v2 Component to be used from AngularJS v1.
+   */
+  downgradeNg2Component(type: Type): Function;
+  /**
+   * Bootstrap a hybrid AngularJS v1 / Angular v2 application.
+   */
+  bootstrap(element: Element, modules?: any[], config?: angular.IAngularBootstrapConfig): UgradeAdapterRef;
+  /**
+   * Adds a provider to the top level environment of a hybrid AngularJS v1 / Angular v2 application.
+   */
+  addProvider(provider: Type | any[] | any): void;
+  /**
+   * Allows Angular v2 service to be accessible from AngularJS v1.
+   */
+  downgradeNg2Provider(token: any): Function;
```

Comment:
we also need to support `upgradeNg1Provider` it will be needed if someone uses legacy ng core services (`$location`...) or so
https://github.com/angular/angular/blob/master/modules/%40angular/upgrade/src/upgrade_adapter.ts#L476

nice thing is they added `asToken` option so ng2 service doesn't need to use `@Inject` with string token